lib/docker/connection.rb

```diff
@@ -38,17 +38,17 @@ def request(*args, &block)
     request = compile_request_params(*args, &block)
     log_request(request)
     resource.request(request).body
-  rescue Excon::Errors::BadRequest => ex
+  rescue Excon::Error::BadRequest => ex
     raise ClientError, ex.response.body
-  rescue Excon::Errors::Unauthorized => ex
+  rescue Excon::Error::Unauthorized => ex
     raise UnauthorizedError, ex.response.body
-  rescue Excon::Errors::NotFound => ex
+  rescue Excon::Error::NotFound => ex
     raise NotFoundError, ex.response.body
-  rescue Excon::Errors::Conflict => ex
+  rescue Excon::Error::Conflict => ex
     raise ConflictError, ex.response.body
-  rescue Excon::Errors::InternalServerError => ex
+  rescue Excon::Error::InternalServerError => ex
```

Comment:
This should be Excon::Errors::InternalServerErrorError (yep, believe it or not)

https://github.com/excon/excon/commit/13758ec567bc090f4fba6e433f191fb92434637c